src/vs/workbench/contrib/debug/browser/disassemblyView.ts

```diff
@@ -522,23 +592,68 @@ class InstructionRenderer extends Disposable implements ITableRenderer<IDisassem
 	}
 
 	renderTemplate(container: HTMLElement): IInstructionColumnTemplateData {
+		const sourcecode = append(container, $('.sourcecode'));
 		const instruction = append(container, $('.instruction'));
+		this.applyFontInfo(sourcecode);
 		this.applyFontInfo(instruction);
-
 		const currentElement: { element?: IDisassembledInstructionEntry } = { element: undefined };
 
 		const disposables = [
-			this._disassemblyView.onDidChangeStackFrame(() => this.rerenderBackground(instruction, currentElement.element))
+			this._disassemblyView.onDidChangeStackFrame(() => this.rerenderBackground(instruction, sourcecode, currentElement.element)),
+			addStandardDisposableListener(sourcecode, 'click', () => this.openSourceCode(currentElement.element?.instruction!)),
 		];
 
-		return { currentElement, instruction, disposables };
+		return { currentElement, instruction, sourcecode, disposables };
 	}
 
 	renderElement(element: IDisassembledInstructionEntry, index: number, templateData: IInstructionColumnTemplateData, height: number | undefined): void {
-		templateData.currentElement.element = element;
+		this.renderElementInner(element, index, templateData, height);
+	}
 
+	private async renderElementInner(element: IDisassembledInstructionEntry, index: number, templateData: IInstructionColumnTemplateData, height: number | undefined): Promise<void> {
+		templateData.currentElement.element = element;
 		const instruction = element.instruction;
-		const sb = createStringBuilder(10000);
+		templateData.sourcecode.innerText = '';
+		const sb = createStringBuilder(1000);
+
+		if (this._disassemblyView.isSourceCodeRender && instruction.location?.path && instruction.line) {
+			let sourceURI = this.getUriFromSource(instruction);
+
+			if (sourceURI) {
+				let textModel: ITextModel | undefined = undefined;
+				const sourceSB = createStringBuilder(10000);
+
+				try {
+					const ref = await this._disassemblyView.textModelService.createModelReference(sourceURI);
+					textModel = ref.object.textEditorModel;
+					templateData.disposables.push(ref);
+
+				} catch {
+					const textFileContent = await this._disassemblyView.nativeTextFileService.read(sourceURI);
+					textModel = new TextModel(textFileContent.value, TextModel.DEFAULT_CREATION_OPTIONS, null, null, this._disassemblyView.undoRedoService);
+				}
+
+				if (textModel) {
+					let lineNumber = instruction.line;
+
+					while (lineNumber && lineNumber >= 1 && lineNumber <= textModel.getLineCount()) {
+						const lineContent = textModel.getLineContent(lineNumber);
+						sourceSB.appendASCIIString(`\u00A0\u00A0${lineNumber}:\u00A0`);
+						sourceSB.appendASCIIString(lineContent.replace(/[ ]/g, '\u00A0') + '\n');
```

Comment:
If you are trying to keep spaces from collapsing, you can do that with CSS. Something like `white-space: pre`: https://developer.mozilla.org/en-US/docs/Web/CSS/white-space

Reply:
> > ![Sep-21-2021 15-25-50](https://user-images.githubusercontent.com/323878/134258635-cbd9f3c8-c181-4ed8-8a42-25d6f4070a0f.gif)
> > horizontal scrolling is not working correctly for me
> 
> @yuehuang010 Horizontal scrolling has never been supported by WorkbenchTable. Please disable it.

My bad. I was testing it and I thought I reverted the change.

Reply:
Now using 'pre'